packages/analytics-react-native/android/src/main/java/com/amplitude/reactnative/AmplitudeReactNativeModule.kt

```diff
@@ -42,4 +50,56 @@ class AmplitudeReactNativeModule(private val reactContext: ReactApplicationConte
             putString("appSetId", androidContextProvider!!.appSetId)
         })
     }
+
+    @ReactMethod
+    private fun getLegacySessionData(instanceName: String?, promise: Promise) {
+        val storage = LegacyDatabaseStorageProvider.getStorage(reactContext.applicationContext, instanceName)
+        val deviceId = storage.getValue(DEVICE_ID_KEY)
+        val userId = storage.getValue(USER_ID_KEY)
+        val previousSessionId = storage.getLongValue(PREVIOUS_SESSION_ID_KEY)
+        val lastEventTime = storage.getLongValue(LAST_EVENT_TIME_KEY)
+        val lastEventId = storage.getLongValue(LAST_EVENT_ID_KEY)
+        promise.resolve(WritableNativeMap().apply {
+            if (deviceId != null) {
+                putString("deviceId", deviceId)
+            }
+            if (userId != null) {
+                putString("userId", userId)
+            }
+            if (previousSessionId != null) {
+                putDouble("sessionId", previousSessionId.toDouble())
+            }
+            if (lastEventTime != null) {
+                putDouble("lastEventTime", lastEventTime.toDouble())
+            }
+            if (lastEventId != null) {
+                putDouble("lastEventId", lastEventId.toDouble())
+            }
+        })
+    }
+
+    @ReactMethod
+    private fun getLegacyEvents(instanceName: String?, eventKind: String, promise: Promise) {
```

Comment:
Can we have a better name for the input string instead of eventKind? What dose that mean?

Reply:
I added the parameter to not add functions like getLegacyEvents/getLegacyIdentifies/getLegacyInterceptedIdentifies, removeLegacyEvent/removeLegacyIdentify/getLegacyInterceptedIdentify
https://github.com/amplitude/Amplitude-TypeScript/blob/4ea90ed2640f822fb203b57a0745716834d9c2b6/packages/analytics-react-native/src/migration/remnant-data-migration.ts#L5
Please, advice better name.

Reply:
Thanks @falconandy that makes sense to me. Since this is only in the migration logic (not seen by customer) I think naming is ok as-is.